packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -67,4 +70,33 @@ export class BlobManager {
 
         return handle;
     }
+
+    public addBlobId(blobId: string) {
+        this.blobIds.add(blobId);
+    }
+
+    /**
+     * Load a set of previously attached blob IDs from a previous snapshot. Note
+     * that BlobManager tracking and reporting attached blobs is a temporary
+     * solution since storage expects attached blobs to be reported and any that
+     * are not reported as attached may be GCed. In the future attached blob
+     * IDs will be collected at summarization time, and runtime will not care
+     * about the existence or specific formatting of this tree in returned
+     * snapshots.
+     *
+     * @param blobsTree - Tree containing IDs of previously attached blobs. This
+     * corresponds to snapshot() below. We look for the IDs in the blob entries
+     * of the tree since the both the r11s and SPO drivers replace the
+     * attachment types returned in snapshot() with blobs.
+     */
+    public load(blobsTree?: ISnapshotTree): void {
+        if (blobsTree) {
+            Object.values(blobsTree.blobs).map((entry) => this.addBlobId(entry));
```

Comment:
this.blobIds should be empty when load happens, is there more efficient way to populate this.blobIds from blobsTree.blobs? I.e. not one by one, but via one call?
I think (correct if I'm wrong, and not sure if it helps with writing more efficient code) it can be an array, not a set, because we should never see duplicate IDs being added to it.

Reply:
>I think (correct if I'm wrong, and not sure if it helps with writing more efficient code) it can be an array, not a set, because we should never see duplicate IDs being added to it.

In the case of r11s, uploading the same image/file twice would generate two blobAttach ops with the same ID. I'm also not sure whether or not attaching one BlobHandle twice will generate two or not. BlobManager does not prevent duplicate blobAttach ops in any case.

In regards to efficiency, we could do `this.blobIds = new Set(Object.values(blobsTree.blobs))`, but I'm not sure if this would have a significant performance impact. We could also delay loading until snapshot() is called (not sure about the performance impact of this either, but it would only happen on summarizing container then).